src/interpreter/inner.rs

```diff
@@ -216,47 +228,36 @@ pub(super) fn from_txdata<'txin>(
                     None, // Tr script code None
                 ))
             } else {
-                // wit_stack.len() >=2
-                // Check for annex
                 let ctrl_blk = wit_stack.pop().ok_or(Error::UnexpectedStackEnd)?;
```

Comment:
nit: mild preference to have an explicity match over wit_stack.len() for 0, 1, _ and return Error::UnexpectedStackEnd for 0, and then make a new error Errror::ImplementationInvariantBug or something for the pops here (so that we're not using expect or unwrap) -- for readability it makes it more clear to me.

Reply:
Changed to match over len(). 

I don't see any expects or unwraps in this code? Where are we using those?

Reply:
we're not, i was just saying that now the ok_or is guaranteed to succeed, so it might make sense to make it an assertion via unwrap or expect (since it should be unreachable) or to add a Error::ImplementationError instead of unwraps or UnexpectedStackEnd (because the user should know they didn't do submit something invalid if it ever hits).